core/trino-spi/src/main/java/io/trino/spi/connector/Redirection.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.trino.spi.predicate.TupleDomain;
+
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class Redirection
```

Comment:
Should this be an inner class of TableScanRedirectApplicationResult ?

Reply:
Yes, let's make it an inner class. The concept of "Redirection" as a first class entity doesn't make sense -- this is just a table + filter. "Redirection" is just the way such an entity is used in the context of the redirection API

Reply:
made this inner class